functions/go/set-image/image_transformer.go

```diff
@@ -114,62 +109,72 @@ func (si *SetImage) validateInput() error {
 }
 
 // Config initializes SetImage from a functionConfig sdk.KubeObject
-func (si *SetImage) Config(functionConfig *sdk.KubeObject) error {
+func (si *SetImage) Config(functionConfig *fn.KubeObject) (error, bool) {
 	si.Image = types.Image{}
 	si.AdditionalImageFields = nil
 	switch {
-	case validGVK(functionConfig, "v1", "ConfigMap"):
-		if found, err := functionConfig.Get(&si.Image, "data"); err != nil {
-			return fmt.Errorf("unable to convert functionConfig to v1 ConfigMap:\n%w", err)
-		} else if !found {
-			return fmt.Errorf("unable to get field data from functionConfig")
-		}
-	case validGVK(functionConfig, fnConfigAPIVersion, fnConfigKind):
-		if err := functionConfig.As(si); err != nil {
-			return fmt.Errorf("unable to convert functionConfig to %s %s:\n%w",
-				fnConfigAPIVersion, fnConfigKind, err)
-		}
+	case functionConfig.IsGVK( "v1", "ConfigMap"):
+		functionConfig.GetOrDie(&si.Image, "data")
+	case functionConfig.IsGVK(fnConfigAPIVersion, fnConfigKind):
+		functionConfig.AsOrDie(si)
 	default:
-		return fmt.Errorf("`functionConfig` must be a `ConfigMap` or `%s`", fnConfigKind)
+		return fmt.Errorf("`functionConfig` must be a `ConfigMap` or `%s`", fnConfigKind), false
 	}
 	if err := si.validateInput(); err != nil {
-		return err
+		return err, false
 	}
 	defaultImageFields, err := getDefaultImageFields()
 	if err != nil {
-		return err
+		return err, false
 	}
 	si.AdditionalImageFields = append(si.AdditionalImageFields, defaultImageFields...)
-	return nil
+	return nil, true
 }
 
 // Transform set image out of place and returns a new []*sdk.KubeObject
```

Comment:
It seems this comment is not accurate now since the input is modified in place

Reply:
ah, good catch! Thanks